### PR TITLE
Added ContentProvider to resolve news models to URLs

### PR DESCRIPTION
**Pull request by aschempp**
This will allow the router to generate URLs for news models.

**Dependencies:**
- [ ] https://github.com/contao/contao/pull/2031

### CODE REVIEW

File: news-bundle/src/Resources/contao/classes/News.php [hunk 1]
```diff
@@ -359,79 +356,31 @@ public function getSearchablePages($arrPages, $intRoot=0, $blnIsSitemap=false)
 	 * @param boolean   $blnAbsolute
 	 *
 	 * @return string
+	 *
+	 * @deprecated Deprecated since Contao 4.10. Use the Symfony router instead.
 	 */
 	public static function generateNewsUrl($objItem, $blnAddArchive=false, $blnAbsolute=false)
 	{
-		$strCacheKey = 'id_' . $objItem->id . ($blnAbsolute ? '_absolute' : '');
+		@trigger_error(__METHOD__.' is deprecated since Contao 4.10. Use the Symfony router instead.', E_USER_DEPRECATED);
```
Comment (bytehead) [inline news-bundle/src/Resources/contao/classes/News.php]: Use `trigger_deprecation` here (see #1946).

**Review by bytehead — CHANGES_REQUESTED** [review]

File: news-bundle/src/Resources/contao/classes/News.php (hunk 1 above)
Comment (aschempp) [reply news-bundle/src/Resources/contao/classes/News.php]: Fixed in 5db43fe2d7f72fb92f3e92804f74419d6fe17ba8

**Comment by bytehead** [comment]
What about tests for the routing provider?

**Comment by leofeyer** [comment]
It seems you forgot to adjust the `PreviewUrlConverterListenerTest` class. And you should add a unit test for the `NewsRouteProvider` class. 🙈

**Pull request closed without merging**